<?xml version="1.0"?>

<launch>
  <arg name="source_bagfile"/>
  <arg name="output_dir"/>
  <arg name="raw"/>
  <arg name="datapoint"/>

  <include file="$(find droplet_underwater_assembly)/launch/description.launch" />
  <node pkg="rosbag" type="play" name="player" required="true" output="screen" args="--start=240 --clock -r 1.0 --duration=90 $(arg source_bagfile)"/>

  <node name="camera_info_publisher" output="screen" pkg="droplet_underwater_assembly" type="yaml_to_camera_info_publisher.py" args="$(find downward_camera_localizer)/param/ost.yaml"/>
      <!--<node name="csv_recorder" output="screen" pkg="droplet_underwater_assembly" type="location_stream_position_recorder.py" args="$(arg datapoint) $(arg output_dir) $(arg raw)"/>-->

  <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/camera_array/cam1/image_raw raw out:=/image_raw"/>

  <node pkg="nodelet" type="nodelet" name="bluerov_nodelet_manager" args="manager" output="screen" respawn="true">
    <param name="num_worker_threads" value="1" />
  </node>

  <node pkg="nodelet" type="nodelet" name="stag_ros" output="screen" args="load stag_ros/StagNodelet bluerov_nodelet_manager" respawn="false" required="false">
    <param name="camera_image_topic" type="string" value="/image_raw" />
    <param name="camera_info_topic" type="string" value="/camera_info" />
    <param name="tag_id_type" type="int" value="21" />
    <param name="marker_frame_prefix" type="string" value="/ar_marker_" />
    <param name="default_marker_size" type="double" value="0.100" />
    <param name="output_frame_id"     type="string" value="/base_link" />
    <param name="image_frame_id"      type="string" value="/cam_0_optical_frame" />
    <param name="marker_message_topic" type="string" value="/bluerov_controller/ar_tag_detector_2" />
    <param name="process_images_in_parallel" type="bool" value="true" />
    <param name="use_marker_bundles" type="bool" value="true" />
    <param name="use_marker_bundles" type="bool" value="false" />
    <param name="publish_debug_images" type="bool" value="true" />
    <rosparam command="load" file="$(find droplet_underwater_assembly)/param/bundle-four-stags-aluminum.json" />
    <rosparam command="load" param="marker_sizes_by_id" file="$(find droplet_underwater_assembly)/param/marker_sizes_by_id.yaml" />
    <param name="track_markers" type="bool" value="true" />
    <param name="marker_track_height_offset" type="int" value="50" />
    <param name="marker_track_width_offset" type="int" value="50" />
    <param name="track_markers" type="bool" value="false" />
    <param name="is_fisheye" type="bool" value="true" />
    <param name="marker_track_height_offset" type="int" value="50" />
    <param name="marker_track_width_offset" type="int" value="50" />
  </node>

  <node pkg="rosbag" type="record" name="recorder" output="screen" args="-o /home/sam/ros-catkin-workspace/data_from_loc_tube_fiducials/with_stag_added /camera_array/cam1/image_raw/compressed /mavros/imu/static_pressure /mavros/imu/data_raw /bluerov_controller/ar_tag_detector_2" />
</launch>

